lib/beaker-puppet/install_utils/puppet_utils.rb

```diff
@@ -90,41 +90,51 @@ def get_puppet_collection(agent_version = 'latest')
           collection
         end
 
-        # Determine the puppet collection that matches a given version of the puppet-agent
-        # package (you can find this version in the `aio_agent_version` fact).
+        # Determine the puppet collection that matches a given package version. The package
+        # must be one of
+        #   * :puppet_agent (you can get this version from the `aio_agent_version_fact`)
+        #   * :puppet
         #
-        # @param agent_version [String] a semver version number of the puppet-agent package, or the string 'latest'
+        #
+        # @param package [Symbol] the package name. must be one of :puppet_agent, :puppet
+        # @param version [String] a version number or the string 'latest'
         # @returns [String|nil] the name of the corresponding puppet collection, if any
-        def puppet_collection_for_puppet_agent_version(agent_version)
-          agent_version = agent_version.to_s
-          return 'puppet' if agent_version.strip == 'latest'
+        def puppet_collection_for(package, version)
+          valid_packages = [
+            :puppet_agent,
+            :puppet
+          ]
 
-          x, y, z = agent_version.to_s.split('.').map(&:to_i)
-          return nil if x.nil? || y.nil? || z.nil?
+          unless valid_packages.include?(package)
+            raise "package must be one of #{valid_packages.join(', ')}"
+          end
 
-          return 'pc1' if x == 1
+          case package
+          when :puppet_agent
+            agent_version = version.to_s
+            return 'puppet' if agent_version.strip == 'latest'
 
-          # A y version >= 99 indicates a pre-release version of the next x release
-          x += 1 if y >= 99
-          "puppet#{x}" if x > 4
-        end
+            x, y, z = agent_version.to_s.split('.').map(&:to_i)
+            return nil if x.nil? || y.nil? || z.nil?
 
-        # Determine the puppet collection that matches a given version of the puppet gem.
-        #
-        # @param version [String] a semver version number of the puppet gem, or the string 'latest'
-        # @returns [String|nil] the name of the corresponding puppet collection, if any
-        def puppet_collection_for_puppet_version(puppet_version)
-          puppet_version = puppet_version.to_s
-          return 'puppet' if puppet_version.strip == 'latest'
+            return 'pc1' if x == 1
+
+            # A y version >= 99 indicates a pre-release version of the next x release
+            x += 1 if y >= 99
+            "puppet#{x}" if x > 4
+          when :puppet
+            puppet_version = version.to_s
+            return 'puppet' if puppet_version.strip == 'latest'
 
-          x, y, z = puppet_version.to_s.split('.').map(&:to_i)
-          return nil if x.nil? || y.nil? || z.nil?
+            x, y, z = puppet_version.to_s.split('.').map(&:to_i)
+            return nil if x.nil? || y.nil? || z.nil?
 
-          return 'pc1' if x == 4
+            return 'pc1' if x == 4
```

Comment:
Is beaker-puppet used int he puppet_agent module? That's a little scary

Reply:
oh wait, I shouldn't be surprised by this. The beaker puppet install helper uses beaker-puppet

Reply:
However, wouldn't this only be needed if we were installing new package? This should be verified, but I'm not convinced we need this for packages that already exist on the system we are upgrading. Just a thought. It would be nice to start removing all the `pc1` references in this code. It's totes something we can save for later.

Reply:
We do need PC1 in the `:puppet_agent` package case because the puppet_agent module tests calculate the collection from the agent version. I don't know about PC1 in the `:puppet` package case though. @caseywilliams ?

Reply:
👍 sounds good :) Safest to keep it